stage1/init/pod.go

```diff
@@ -507,45 +491,29 @@ func (p *Pod) appToNspawnArgs(ra *schema.RuntimeApp) ([]string, error) {
 	if err := os.Chmod(sharedVolPath, sharedVolPerm); err != nil {
 		return nil, fmt.Errorf("could not change permissions of %q: %v", sharedVolPath, err)
 	}
-	// Here we bind the volumes to the mountpoints via runtime mounts (--mount)
+
+	vols := make(map[types.ACName]types.Volume)
 	for _, v := range p.Manifest.Volumes {
 		vols[v.Name] = v
-		if v.Kind == "empty" {
-			if err := os.MkdirAll(filepath.Join(sharedVolPath, v.Name.String()), sharedVolPerm); err != nil {
-				return nil, fmt.Errorf("could not create shared volume %q: %v", v.Name, err)
-			}
-		}
-	}
 
-	for _, mp := range app.MountPoints {
-		// there's already an injected mount for this target path, skip
-		if _, ok := mounts[mp.Path]; ok {
-			continue
+		if err := os.MkdirAll(filepath.Join(sharedVolPath, v.Name.String()), sharedVolPerm); err != nil {
```

Comment:
Looks like the directory is created for all volumes, not only the "empty" ones. Intended?